lib/main.dart

```diff
@@ -2,23 +2,54 @@ import 'package:flutter/material.dart';
 import 'package:off_top_mobile/login.dart';
 import 'package:off_top_mobile/routing/router.dart' as router;
 import 'package:off_top_mobile/routing/routing_constants.dart';
+import 'package:off_top_mobile/themes/themeProvider.dart';
+import 'package:off_top_mobile/themes/themeSelector.dart';
+import 'package:provider/provider.dart';
 import 'package:syncfusion_flutter_core/core.dart';
 
 void main() {
   SyncfusionLicense.registerLicense(
       'NT8mJyc2IWhia31ifWN9Z2FoZnxha3xhY2Fjc2JiaWZlaWdlcwMeaDImOzE8PRM0PjI6P30wPD4=');
+
   runApp(MyApp());
 }
 
-class MyApp extends StatelessWidget {
+class MyApp extends StatefulWidget {
+  @override
+  _MyAppState createState() => _MyAppState();
+}
+
+class _MyAppState extends State<MyApp> {
+  final ThemeProvider themeChangeProvider = ThemeProvider();
+
+  @override
+  void initState() {
+    super.initState();
+    getCurrentAppTheme();
+  }
+
+  Future<void> getCurrentAppTheme() async {
+    themeChangeProvider.theme =
+        await themeChangeProvider.themePreference.getTheme();
+  }
+
   @override
   Widget build(BuildContext context) {
-    return const MaterialApp(
-      /*onGenerateRoute: router.generateRoute,
-      title: 'Login Page',
-      initialRoute: LoginRoute,*/
-      title: 'Login Page',
-      home: LoginPage(), //changed route to the login page as it's home page
+    return ChangeNotifierProvider(
+      create: (_) {
+        return themeChangeProvider;
+      },
+      child: Consumer<ThemeProvider>(
+        builder: (BuildContext context, ThemeProvider value, Widget child) {
+          return MaterialApp(
+            debugShowCheckedModeBanner: false,
+            title: 'Login Page',
+            theme: ThemeSelector.themeData(themeChangeProvider.theme, context),
+            home:
```

Comment:
now you can pass in the values here and you will no longer need to worry about the if-checks

`themData(themeChangeProvider.primaryColor, themeChangeProvider.secondaryColor, ...)`